packages/legacy-client/src/client/objectSets/OsdkObjectSet.test.ts

```diff
@@ -257,6 +257,32 @@ describe("OsdkObjectSet", () => {
     expect(result.type).toEqual("ok");
   });
 
+  it("supports round-trip of circular links", async () => {
+    const os = createBaseTodoObjectSet(client);
+    mockObjectPage([mockTodoObject]);
+    const todoResponse = await os.get("1");
+
+    if (!isOk(todoResponse)) {
+      expect.fail("todo response was not ok");
+    }
+
+    mockObjectPage([mockTaskObject]);
+    const taskResponse = await todoResponse.value.linkedTask.get();
+
+    if (!isOk(taskResponse)) {
+      expect.fail("task response was not ok");
+    }
+
+    mockObjectPage([mockTodoObject]);
+    const linkedTodosResponse = await taskResponse.value.linkedTodos.all();
+
+    if (!isOk(linkedTodosResponse)) {
+      expect.fail("linked todos response was not ok");
+    }
+
+    expect(linkedTodosResponse.value.length).toEqual(1);
+  });
```

Comment:
I'm kind of sad that this wound up in the OsdkObjectSet.test.ts file, but the scaffolding for getting objects and handling network requests is pretty nice. Thoughts?

Reply:
We can probably move all the scaffolding into the test utils folder we have

Reply:
I made an attempt at breaking things up to make them reusable (mostly just pushing the mock data into the MockOntology, TodoObject, TaskObject files) and it seemed like more work than it was worth. I'm going to leave it as-is for now.